lib/posix/spawn/child.rb

```diff
@@ -196,16 +209,16 @@ def exec!
       #   exceeds the amount specified by the max argument.
       def read_and_write(input, stdin, stdout, stderr, timeout=nil, max=nil)
         max = nil if max && max <= 0
-        @out, @err = '', ''
 
         # force all string and IO encodings to BINARY under 1.9 for now
-        if @out.respond_to?(:force_encoding) and stdin.respond_to?(:set_encoding)
+        if @stdout_buffer.respond_to?(:set_encoding)
+          bin_encoding = Encoding::BINARY
           [stdin, stdout, stderr].each do |fd|
-            fd.set_encoding('BINARY', 'BINARY')
+            fd.set_encoding(bin_encoding, bin_encoding)
```

Comment:
Can we do this on a per `fd` basis?
```ruby
bin_encoding = Encoding::BINARY
[stdin, stdout, stderr].each do |fd|
  fd.set_encoding(bin_encoding, bin_encoding) if fd.respond_to?(:set_encoding)
end
```

Also, are we intentionally dropping the `force_encoding` calls on `stdout` and `stderr` here?